### PR TITLE
fix: cleanup in circulating supply

### PR DESCRIPTION
**Pull request by telezhnaya**
move run_circulating_supply_computation to mod rs
prepare the place for other aggregated modules
get rid of cargo clippy warnings

### CODE REVIEW

File: src/aggregated/circulating_supply/lockup_types.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#![allow(clippy::all)]
```
Comment (khorolets) [inline src/aggregated/circulating_supply/lockup_types.rs:1]: is it really necessary here?

**Review by khorolets — APPROVED** [review]
LGTM

File: src/aggregated/circulating_supply/lockup_types.rs (hunk 1 above)
Comment (telezhnaya) [reply src/aggregated/circulating_supply/lockup_types.rs:1]: This file gives ~10 warnings. I don't want to touch copy-pasted code (and, to be honest, I don't know how to fix some of these warnings properly 😄)

Comment (khorolets) [reply src/aggregated/circulating_supply/lockup_types.rs:1]: Could you share some of them so we can have a look?

cc @frol

Comment (telezhnaya) [reply src/aggregated/circulating_supply/lockup_types.rs:1]: 11, and they all go from `construct_uint!`

```
warning: manual `RangeInclusive::contains` implementation
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = note: `#[warn(clippy::manual_range_contains)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_range_contains
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: manual implementation of an assign operation
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = note: `#[warn(clippy::assign_op_pattern)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assign_op_pattern
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: manual implementation of an assign operation
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assign_op_pattern
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: manual implementation of an assign operation
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assign_op_pattern
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: manual implementation of an assign operation
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assign_op_pattern
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of `offset` with a `usize` casted to an `isize`
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = note: `#[warn(clippy::ptr_offset_with_cast)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_offset_with_cast
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of `offset` with a `usize` casted to an `isize`
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_offset_with_cast
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of `offset` with a `usize` casted to an `isize`
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_offset_with_cast
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of `offset` with a `usize` casted to an `isize`
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_offset_with_cast
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of `offset` with a `usize` casted to an `isize`
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_offset_with_cast
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: manual implementation of an assign operation
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assign_op_pattern
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: 11 warnings emitted

```

Comment (frol) [reply src/aggregated/circulating_supply/lockup_types.rs:1]: The warnings come from uint macro, so we cannot really do anything about it. Let's at least narrow the `allow` list down.

```suggestion
#![allow(clippy::ptr_offset_with_cast, clippy::assign_op_pattern)]
```

File: src/main.rs [hunk 2]
```diff
@@ -278,15 +277,7 @@ fn main() {
...
-
-                if indexer.near_config().genesis.config.chain_id == "mainnet" {
-                    actix::spawn(
-                        circulating_supply_provider::run_circulating_supply_computation(
-                            view_client,
-                            pool,
-                        ),
-                    );
-                }
+                aggregated::spawn_aggregated_computations(view_client, pool, &indexer);
```
Comment (frol) [inline src/main.rs:280]: Given that you get `view_client` from `indexer`, I don't see a need to pass `view_client` separately to the `spawn_aggregated_computations`

**Review by frol — APPROVED** [review]

**Pull request merged**